### ISSUE TITLE
Training Loop (scripts/train.py)

### ISSUE THREAD

**Issue by wecncodecrew** [wecncodecrew/La-Lango#24]
Implement the PyTorch training loop (forward pass, loss calculation, backward pass, optimizer step).

**Comment by reuben-vitalis** [comment]
i will work this out - reuben vitalis
